lib/tapioca/dsl/compilers/active_record_fixtures.rb

```diff
@@ -190,10 +196,14 @@ def fixture_class_mapping_from_fixture_files
                   next unless ::File.file?(file)
 
                   ActiveRecord::FixtureSet::File.open(file) do |fh|
+                    fixture_name = file.delete_prefix(path.to_s).delete_prefix("/").delete_suffix(".yml")
                     next unless fh.model_class
 
-                    fixture_name = file.delete_prefix(path.to_s).delete_prefix("/").delete_suffix(".yml")
                     mapping[fixture_name] = fh.model_class
+                  rescue ActiveRecord::Fixture::FormatError
+                    # For fixtures that are not associated to any models and just contain raw data or fixtures that
+                    # contain invalid formatting, we want to skip them and avoid crashing
+                    method_names.delete(T.must(fixture_name))
```

Comment:
Hmm, I am not sure I really like the fact that we are modifying the array we are iterating on when we've called this method here.

I'd rather have us filter invalid fixtures way earlier in the process, even if it means iterating on them twice.

Alternatively, we could turn this method into something that builds the methods we should be iterating on or something.

Reply:
I switched to filtering ahead of time. We already had a map of `fixture_name => Model`, so we can use that to filter.